src/drivers/usb/usb_pnp.c

```diff
@@ -77,15 +85,28 @@ void usb_dev_addr_settled(struct usb_dev *dev) {
 	usb_dev_configure(dev);
 }
 
+static int usb_add_newdev_action(struct lthread *self) {
+	struct usb_dev *dev;
+
+	dlist_foreach_entry(dev, &usb_newdev_list, dev_link) {
+		dlist_del(&dev->dev_link);
+		dlist_head_init(&dev->dev_link);
```

Comment:
Hm, I looked at src/util/dlist/dlist_debug.h: dlist_del_init and what I see is that dlist_del_init calls dlist_init after dlist_del, not dlist_head_init.